### ISSUE TITLE
Nullability analysis in constructor - no warning if autoproperty assigned to itself

### ISSUE THREAD

**Issue by lellid** [dotnet/roslyn#47490]
**16.8.0preview2.1 / net5.0preview8**: 

**Steps to Reproduce**:

No error or warning is issued if in the constructor property `A` is assigned to itself (in this case it was a typo- silly me!):
```
#nullable enable
    class Foo
    {
        object A { get; set; }
        public Foo(object a)
        {
            A = A;
        }
    }
```
If `A` would be a field instead of an autoproperty, a warning is issued (but not about nullability).

But consider the next example:

```
#nullable enable
    class Foo
    {
        object A;
        object B;

        public Foo(object a)
        {
            A = B;
            B = A;
        }
    }

```

Again no error or warning, even if `A` and `B` are fields!

**Expected Behavior**:

A warning should be issued that an uninitialized member is used.

**Actual Behavior**:

No warning or error.

**Comment by RikkiGibson** [comment]
I believe this was fixed in #46441. Please let us know if you can still reproduce after VS 16.8-preview3 comes out.

**Comment by RikkiGibson** [comment]
Although, I believe that there is a more general warning that we may want in a warning wave about assigning a property to itself (perhaps in any context).
